rbi/annotations/shopify-money.rbi

```diff
@@ -0,0 +1,173 @@
+# typed: strong
+
+class Money
+  sig { returns(BigDecimal) }
+  attr_reader :value
+
+  sig { returns(T.any(Money::Currency, Money::NullCurrency)) }
+  attr_reader :currency
+
+  sig do
+    params(
+      value: T.nilable(T.any(Money, Numeric, String)),
+      currency: T.nilable(T.any(Money::Currency, Money::NullCurrency, String)),
+    )
+      .void
+  end
+  def initialize(value, currency); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(T::Boolean) }
+  def zero?(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(T::Boolean) }
+  def nonzero?(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(T::Boolean) }
+  def positive?(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(T::Boolean) }
+  def negative?(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(Integer) }
+  def to_i(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(Float) }
+  def to_f(*args, **_arg1, &block); end
+
+  # @method_missing: delegated to BigDecimal
+  sig { params(args: T.untyped, _arg1: T.untyped, block: T.nilable(T.proc.void)).returns(Integer) }
+  def hash(*args, **_arg1, &block); end
+
+  class << self
+    sig { params(block: T.nilable(T.proc.params(config: Money::Config).void)).void }
+    def configure(&block); end
+
+    sig do
+      params(
+        value: T.nilable(T.any(Money, Numeric, String)),
+        currency: T.nilable(T.any(Money::Currency, Money::NullCurrency, String)),
+      )
+        .returns(Money)
```

Comment:
This is now causing errors when `Money` instances are multiplied with integers, `5 * Money.new` as the second argument isn't an Integer https://github.com/sorbet/sorbet/blob/dd4602e030832c930d0b413f31734a83708448f6/rbi/core/integer.rbi#L59.

Reply:
Ah, yeah, that's a bummer. I can't think of an easy fix though -- we'd need to be able to overload the signatures for `Integer#*`, `Float#*`, etc. with an additional signature, and I think Sorbet only supports overloading for its own stdlib types.

Maybe this is an acceptable limitation? It's easily fixed by swapping the arguments so that `Money#*` is called instead. It would be nice if this could be made more obvious but I'm not sure how.

Open to any and all suggestions for improving this :)

Reply:
Yeah we can't represent the [current implementation](https://github.com/Shopify/money/blob/2ebdb97174e743829542b3dafbf6cca42512d084/lib/money/money.rb#L170-L194) without overloading that signature.

I'm okay with swapping the arguments solution. Only problem is it's hard to communicate this moving forward.